server/src/main/java/org/apache/druid/server/metrics/BrokerSegmentCountStatsMonitor.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.server.metrics;
+
+import com.google.inject.Inject;
+import org.apache.druid.java.util.emitter.service.ServiceEmitter;
+import org.apache.druid.java.util.emitter.service.ServiceMetricEvent;
+import org.apache.druid.java.util.metrics.AbstractMonitor;
+import org.apache.druid.server.coordinator.stats.RowKey;
+
+import java.util.Map;
+
+/**
+ * Monitor that tracks the number of segments of a datasource currently queryable by this Broker.
+ */
+public class BrokerSegmentCountStatsMonitor extends AbstractMonitor
```

Comment:
Nit: I was wondering if we shouldn't just call it `BrokerSegmentCountMonitor`.
`*CountStats*` feels a little redundant. You can also rename the provider interface accordingly?

What do you think?

Reply:
Let's call it `BrokerSegmentStatsMonitor`, might make it easy to extend later farther than just counts.